src/physlr-molecules.cc

```diff
@@ -314,35 +324,86 @@ biconnectedComponents(graph_t& subgraph, vertexToComponent_t& vertexToComponent)
 	}
 }
 
+template <class Graph, class vertexIter, class edgeSet>
+void
+make_subgraph(Graph& g, Graph& subgraph, edgeSet& edge_set, vertexIter vBegin, vertexIter vEnd)
+{
+    // //   Make a vertex-induced subgraph of graph g, based on vertices from vBegin to vEnd
+    // //   track the source node by indexOriginal
+
+    // Add vertices into the subgraph, but set `indexOriginal` the index of it in the source graph.
+    for (auto& vIter = vBegin; vIter != vEnd ; ++vIter)
+    {
+        auto u = boost::add_vertex(subgraph);
+
+        subgraph[u].name = g[*vIter].name;
+        subgraph[u].weight = g[*vIter].weight;
+		subgraph[u].indexOriginal = g[*vIter].indexOriginal;
+    }
+
+    // Iterate over all pairs of vertices in the subgraph:
+    // check whether there exist an edge between their corresponding vertices in the source graph.
+    graph_t::vertex_iterator vIter1, vIter2, vend1, vend2;
+    for (boost::tie(vIter1, vend1) = vertices(subgraph); vIter1 != vend1; ++vIter1)
+    {
+        for (boost::tie(vIter2, vend2) = vertices(subgraph); vIter2 != vend2; ++vIter2)
+        {
+            if (vIter1 != vIter2)
+            {
+		        tsl::robin_map<
+		            std::pair<std::size_t,size_t>, int,
+		                boost::hash<std::pair<size_t,size_t>>>::const_iterator got =
+		                edge_set.find(
+                            std::pair<std::size_t,size_t>(
+                                subgraph[*vIter1].indexOriginal,
+                                subgraph[*vIter2].indexOriginal));
```

Comment:
Does this code work? You don't need `std::make_pair<std::size_t,size_t>` ?

Reply:
Yes I believe, as I have tested this, and I do have output of the same size to the original code. I just checked and they have the same number of lines (the `tsv` files).